src/StructuredLogViewer/Controls/TextViewerControl.xaml.cs

```diff
@@ -159,6 +160,63 @@ public void SetText(string text)
                 return;
             }
 
+            Match matches;

```

Comment:
Should this logic perhaps go into BuildControl.xaml.cs -> DisplayText() on line 2371?

Also maybe extract method so that this logic is a separate method.

Reply:
On the other hand perhaps keep it here, maybe in the future we can do smart highlighting to highlight substrings using the textEditor

Reply:
still, I'd extract this logic to a separate method

Reply:
Moved to a separate function.  I wanted to add Outlining to help visualize the position, but it might be too visually busy.  Maybe in the future, it could have toggle button.